x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -8,24 +8,57 @@ import * as React from 'react';
 import ReactDOM from 'react-dom';
 import { CoreStart, AppMountParameters } from 'kibana/public';
 import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
+import { History, createHashHistory } from 'history';
+import { Route, Router, Switch } from 'react-router-dom';
 
 /**
  * This module will be loaded asynchronously to reduce the bundle size of your plugin's main bundle.
  */
 export function renderApp(coreStart: CoreStart, { element }: AppMountParameters) {
   coreStart.http.get('/api/endpoint/hello-world');
 
-  ReactDOM.render(<AppRoot />, element);
+  const history = createHashHistory();
+  ReactDOM.render(<AppRoot history={history} />, element);
 
   return () => {
     ReactDOM.unmountComponentAtNode(element);
   };
 }
 
-const AppRoot = React.memo(() => (
+interface RouterProps {
+  history: History;
+}
+
+const AppRoot: React.FC<RouterProps> = React.memo(({ history }) => (
   <I18nProvider>
-    <h1 data-test-subj="welcomeTitle">
-      <FormattedMessage id="xpack.endpoint.welcomeTitle" defaultMessage="Hello World" />
-    </h1>
+    <Router history={history}>
```

Comment:
Consider using React Router's `HashRouter` instead --- until we can figure out how to take use the Functional test framework with "real" URLs and then switch over to react-router's `BrowserRotuer` component.

Reply:
I could also probably write a helper function to just navigate us to a real url.  Other plugins in Kibana all seem to use hashes and I just wanna understand why.

Reply:
I think the "old" platform client side framework was hash based so that might be the reason why all (most) current functional test use it. 🤷‍♂

Reply:
Now that we figured out support for "real" URLs, consider using react-router's `<BrowserRouter />` component instead, which avoid having to create/manage a `history` instance ourselves